src/pcapi/scripts/suspend_fraudulent_pro_users.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from typing import List
+
+from pcapi.core.bookings.api import cancel_booking_for_fraud
+from pcapi.core.bookings.repository import find_cancellable_bookings_by_offerer
+from pcapi.core.users.api import suspend_account
+from pcapi.core.users.constants import SuspensionReason
+from pcapi.core.users.models import User
+from pcapi.repository.user_offerer_queries import find_one_or_none_by_user_id
+from pcapi.repository.user_queries import find_pro_users_by_email_provider
+from pcapi.scripts.offerer.delete_cascade_offerer_by_id import delete_cascade_offerer_by_id
+
+
+logger = logging.getLogger(__name__)
+
+
+def suspend_fraudulent_pro_by_email_providers(
+    fraudulent_email_providers: List[str], admin_user: User, dry_run: bool = True
+) -> None:
+    fraudulent_pros = []
+
+    for email_provider in fraudulent_email_providers:
+        fraudulent_pros.extend(find_pro_users_by_email_provider(email_provider=email_provider))
+
+    if not dry_run:
+        for fraudulent_pro in fraudulent_pros:
+            try:
+                user_offerrer = find_one_or_none_by_user_id(user_id=fraudulent_pro.id)
+                delete_cascade_offerer_by_id(offerer_id=user_offerrer.offererId)
+            except AttributeError:
+                _cancel_bookings_by_fraudulent_pro(offerer_id=user_offerrer.offererId)
+
+        _suspend_fraudulent_pro_users(users=fraudulent_pros, admin_user=admin_user)
+    else:
+        logger.info("Suspension of users from fraudulent email providers %s", len(fraudulent_pros))
```

Comment:
On pourrait ne pas utiliser le logger car ce message ne sert qu'à la personne qui lance le script. On n'a pas besoin de logger l'info pour l'historique, puisque c'est un _dry run_.

```suggestion
        print("dry run: would suspend %d fraudulent pro user", len(fraudulent_pros))
```

(... De plus, je ne suis pas sûr de l'intérêt du dry run si on n'indique pas tout ce qui serait supprimé : email de l'utilisateur pro, identifiant de l'Offerer et de ses Venue, etc. Mais c'est à voir avec Jean-Baptiste.)

Reply:
J'ai ajouté les e-mails des pros histoire d'avoir plus de feedbacks.